### PR TITLE
PKG-4984 update to 5.7.2

### PR DESCRIPTION
**Pull request by cbouss**
https://github.com/jupyter/jupyter_core/tree/v5.7.2
https://anaconda.atlassian.net/browse/PKG-4984

### CODE REVIEW

**Review by JamesRobertsonGames — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,8 +36,11 @@ requirements:
 test:
   source_files:
     - tests
+    - pyproject.toml
```
Comment (ifitchet) [inline recipe/meta.yaml:39]: Why do you need `pyproject.toml` in tests?

Comment (cbouss) [reply recipe/meta.yaml:39]: pytest reads test configuration from there when available.

Comment (ifitchet) [reply recipe/meta.yaml:39]: Hmm.  So we should copy it for tests if it exists (and possibly some other conditions)?
(Thinking about the code generator)

Comment (cbouss) [reply recipe/meta.yaml:39]: The relevant part here was this:
https://github.com/jupyter/jupyter_core/blob/v5.7.2/pyproject.toml#L112-L129

Sometimes it's worth taking, sometimes not. Sometimes that configuration is found in other files (setup.cfg, conftest.py...) https://docs.pytest.org/en/7.1.x/reference/customize.html

**Pull request merged**